src/rulesets/Base/array.jl

```diff
@@ -278,3 +278,76 @@ function rrule(::typeof(fill), value::Any, dims::Int...)
     end
     return fill(value, dims), fill_pullback
 end
+
+#####
+##### `extrema`, `findmax`, `maximum`, etc.
+#####
+
+function rrule(::typeof(extrema), x::AbstractArray{<:Number}; dims=:)
+    yplus, iplus = findmax(x; dims=dims)
+    yminus, imminus = findmin(x; dims=dims)
+    project = ProjectTo(x)
+    function extrema_pullback((dyplus, dyminus))
+        x_thunk = @thunk begin
+            dx = fill!(similar(x, eltype(dy)), false)
+            view(dx, iplus) .= dyplus
+            view(dx, iminus) .+= dyminus
+            project(dx)
+        end
+        x_ithunk = InplaceableThunk(x_thunk) do dx
+            view(dx, iplus) .+= dyplus
+            view(dx, iminus) .+= dyminus
+            dx
+        end
+        return (NoTangent(), x_ithunk)
+    end
+    return (yplus, yminus), extrema_pullback
+end
+
+function rrule(::typeof(findmax), x::AbstractArray{<:Number}; dims=:)
+    y, ind = findmax(x; dims=dims)
+    project = ProjectTo(x)
+    function findmax_pullback((dy, _))
+        x_thunk = @thunk begin
+            dx = fill!(similar(x, eltype(dy)), false)
+            view(dx, ind) .= dy  # possibly 0-dim view, allows dy::Number and dy::Array cases
```

Comment:
Error on 1.0 from CI:
```
14
WARNING: local variable T conflicts with a static parameter in Type at /home/runner/.julia/packages/ChainRulesCore/ykAd9/src/projection.jl:350.
15
ERROR: LoadError: LoadError: syntax: invalid assignment location "view(dx, iminus)"
16
...
29
in expression starting at /home/runner/work/ChainRules.jl/ChainRules.jl/src/rulesets/Base/array.jl:286
30
in expression starting at /home/runner/work/ChainRules.jl/ChainRules.jl/src/ChainRules.jl:32
31
ERROR: LoadError: Failed to precompile ChainRules [082447d4-558c-5d27-93f4-14fc19e9eca2] to /home/runner/.julia/compiled/v1.0/ChainRules/AzGmQ.ji.
```
Besides which, it would be worth checking whether this trick works on CuArrays

Reply:
Works find for gpu:
```
julia> using CUDA

julia> CUDA.allowscalar(false)

julia> a = cu(rand(3));

julia> view(a,2) .+= 100;

julia> a
3-element CuArray{Float32, 1}:
   0.6533579
 100.13896
   0.037792206

julia> b = cu(rand(4,4));

julia> mx, is = findmax(b, dims=1)
(Float32[0.981492 0.73596996 0.84038144 0.8083734], CartesianIndex{2}[CartesianIndex(3, 1) CartesianIndex(3, 2) CartesianIndex(4, 3) CartesianIndex(1, 4)])

julia> view(b, is) .+= 100;

julia> b
4×4 CuArray{Float32, 2}:
   0.437412    0.174363    0.344238  100.808
   0.591551    0.134952    0.71408     0.61187
 100.981     100.736       0.746843    0.175168
   0.696413    0.207157  100.84        0.235004
```
But not on 1.0:
```
julia> a = [1,2,3];

julia> view(a,1) .+= 4
ERROR: syntax: invalid assignment location "view(a, 1)"

julia> VERSION
v"1.0.5"
```